File: NetFramework/TouchToolkit/VB/TouchToolkitDemo/GcZoomPages/Help UserControl/ZoomControlBar.vb
Imports System.Collections.Generic
Imports System.ComponentModel
Imports System.Drawing
Imports System.Data
Imports System.Linq
Imports System.Text
Imports System.Windows.Forms
Imports C1.Win.TouchToolKit

Namespace C1ZoomPages.Help_Forms
	Public Partial Class ControlBarUserControl02
		Inherits UserControl
		Public Sub New()
			InitializeComponent()
		End Sub

		Protected Overrides Sub OnLoad(e As EventArgs)
            myTrackBar1.Value = Me.gcZoomCommandProvider1.OwnerGcZoom.ZoomFactor
            AddHandler myTrackBar1.ValueChanged, AddressOf myTrackBar1_ValueChanged
            AddHandler Me.gcZoomCommandProvider1.OwnerGcZoom.ZoomFactorChanged, AddressOf OwnerGcZoom_ZoomFactorChanged
            MyBase.OnLoad(e)
        End Sub

        Private Sub OwnerGcZoom_ZoomFactorChanged(sender As Object, e As EventArgs)
            myTrackBar1.Value = Me.gcZoomCommandProvider1.OwnerGcZoom.ZoomFactor
        End Sub

        Private Sub myTrackBar1_ValueChanged(sender As Object, e As EventArgs)
            Me.gcZoomCommandProvider1.OwnerGcZoom.ZoomFactor = myTrackBar1.Value
        End Sub
	End Class
End Namespace

File: NetFramework/WinForms/CS/TouchDemo/ZoomPages/Help UserControl/ZoomControlBar.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Drawing;
using System.Data;
using System.Linq;
using System.Text;
using System.Windows.Forms;
using C1.Win.TouchToolKit;

namespace TouchDemo.C1ZoomPages.Help_Forms
{
    public partial class ControlBarUserControl02 : UserControl
    {
        public ControlBarUserControl02()
        {
            InitializeComponent();
        }

        protected override void OnLoad(EventArgs e)
        {
            myTrackBar1.Value = this.C1ZoomCommandProvider1.OwnerGcZoom.ZoomFactor;
            myTrackBar1.ValueChanged += myTrackBar1_ValueChanged;
            this.C1ZoomCommandProvider1.OwnerGcZoom.ZoomFactorChanged += OwnerC1Zoom_ZoomFactorChanged;
            base.OnLoad(e);
        }

        void OwnerC1Zoom_ZoomFactorChanged(object sender, EventArgs e)
        {
            myTrackBar1.Value = this.C1ZoomCommandProvider1.OwnerGcZoom.ZoomFactor;
        }

        void myTrackBar1_ValueChanged(object sender, EventArgs e)
        {
            this.C1ZoomCommandProvider1.OwnerGcZoom.ZoomFactor = myTrackBar1.Value;
        }
    }
}
